packages/framework/src/Command/FixExtendedClassesAnnotationsCommand.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Command;
+
+use Shopsys\FrameworkBundle\Component\ClassExtension\ClassExtensionRegistry;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Finder\Finder;
+
+class FixExtendedClassesAnnotationsCommand extends Command
+{
+    /**
+     * @var string
+     */
+    protected static $defaultName = 'shopsys:extended-classes:fix-annotations';
+
+    /**
+     * @var string
+     */
+    protected $projectRootDirectory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\ClassExtension\ClassExtensionRegistry
+     */
+    protected $classExtensionRegistry;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Replaces the framework with the project annotations in all project files when there exists a project extension of a given framework class.');
+    }
+
+    /**
+     * @param string $projectRootDirectory
+     * @param \Shopsys\FrameworkBundle\Component\ClassExtension\ClassExtensionRegistry $classExtensionRegistry
+     */
+    public function __construct(string $projectRootDirectory, ClassExtensionRegistry $classExtensionRegistry)
+    {
+        $this->projectRootDirectory = $projectRootDirectory;
+        $this->classExtensionRegistry = $classExtensionRegistry;
+        parent::__construct();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->replaceFrameworkWithProjectAnnotations();
+        $output->writeln('Annotations fixed successfully');
+
+        return 0;
+    }
+
+    protected function replaceFrameworkWithProjectAnnotations(): void
+    {
+        $finder = $this->getFinderForReplacingAnnotations();
+        $annotationsReplacementsMap = $this->getAnnotationsReplacementsMap($this->classExtensionRegistry->getServiceExtensionMap());
+        foreach ($finder as $fileWithFrameworkAnnotationsToReplace) {
+            $pathname = $fileWithFrameworkAnnotationsToReplace->getPathname();
+            $content = file_get_contents($pathname);
+            $replacedContent = preg_replace(
+                array_keys($annotationsReplacementsMap),
+                $annotationsReplacementsMap,
```

Comment:
array_values to be explicit about what is happening

Reply:
fixed and extracted in 788f0c5f5

Reply:
...and fixed in 2f4d122d9 :D